ros/src/common/cmake/autoware_build_flags/cmake/autoware_build_flags-extras.cmake

```diff
@@ -9,4 +9,23 @@ if(${CMAKE_VERSION} VERSION_LESS "3.1.0")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 else()
   set(CMAKE_CXX_STANDARD 11)
-endif()
\ No newline at end of file
+endif()
+
+find_package(CUDA QUIET)
+find_package(Eigen3 QUIET)
+
+option(CPU_ONLY "CPU_Only" TRUE)
+
+####### Melodic Checks
+if (NOT CPU_ONLY AND CUDA_FOUND AND "$ENV{ROS_DISTRO}" STREQUAL "melodic" )
+  if(${CMAKE_VERSION} VERSION_LESS "3.12.3")
+    message(FATAL_ERROR "GPU support on Melodic requires CMake version>= 3.12.3")
+  else()
+    if (${EIGEN3_VERSION_STRING} VERSION_LESS "3.3.6")
+      message(FATAL_ERROR "GPU support on Melodic requires Eigen version>= 3.3.6")
+    endif()
+  endif()
+else()
+  message(WARNING "Building on CPU only mode. Use -DCPU_ONLY=OFF to enable GPU support.")
```

Comment:
If you change the flag name, this message will also need to change to "CUDA support is disabled. Use -DENABLE_CUDA=TRUE and install CMake >= 3.12.3 and Eigen >= 3.3.6 to enable it."

Reply:
Thanks to #2267 We'll just ask for Eigen